SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -789,6 +789,647 @@ void CFlowOutput::Set_CpInverseDesign(CSolver *solver, const CGeometry *geometry
+        EquivArea_PhiAngle[iPhiAngle][iVertex] = 0.0;
+
+        Coord_i = Xcoord_PhiAngle[iPhiAngle][iVertex]*cos(AoA) - Zcoord_PhiAngle[iPhiAngle][iVertex]*sin(AoA);
+
+        for (jVertex = 0; jVertex < iVertex-1; jVertex++) {
+
+          Coord_j = Xcoord_PhiAngle[iPhiAngle][jVertex]*cos(AoA) - Zcoord_PhiAngle[iPhiAngle][jVertex]*sin(AoA);
+          jp1Coord = Xcoord_PhiAngle[iPhiAngle][jVertex+1]*cos(AoA) - Zcoord_PhiAngle[iPhiAngle][jVertex+1]*sin(AoA);
+
+          jFunction = factor*(Pressure_PhiAngle[iPhiAngle][jVertex] - Pressure_Inf)*sqrt(Coord_i-Coord_j);
+          jp1Function = factor*(Pressure_PhiAngle[iPhiAngle][jVertex+1] - Pressure_Inf)*sqrt(Coord_i-jp1Coord);
+
+          DeltaX = (jp1Coord-Coord_j);
+          MeanFuntion = 0.5*(jp1Function + jFunction);
+          EquivArea_PhiAngle[iPhiAngle][iVertex] += DeltaX * MeanFuntion;
+        }
+      }
+    }
+
+    /*--- Create a file with the equivalent area distribution at each azimuthal angle ---*/
+
+    NearFieldEA_file.precision(15);
+
+    if (output) {
+
+      NearFieldEA_file.open("Equivalent_Area.dat", ios::out);
+      NearFieldEA_file << "TITLE = \"Equivalent Area evaluation at each azimuthal angle\"" << "\n";
+
+      if (config->GetSystemMeasurements() == US)
+        NearFieldEA_file << "VARIABLES = \"Height (in) at r="<< R_Plane*12.0 << " in. (cyl. coord. system)\"";
+      else
+        NearFieldEA_file << "VARIABLES = \"Height (m) at r="<< R_Plane << " m. (cylindrical coordinate system)\"";
+
+      for (iPhiAngle = 0; iPhiAngle < PhiAngleList.size(); iPhiAngle++) {
+        if (config->GetSystemMeasurements() == US)
+          NearFieldEA_file << ", \"Equivalent Area (ft<sup>2</sup>), <greek>F</greek>= " << PhiAngleList[iPhiAngle] << " deg.\"";
+        else
+          NearFieldEA_file << ", \"Equivalent Area (m<sup>2</sup>), <greek>F</greek>= " << PhiAngleList[iPhiAngle] << " deg.\"";
+      }
+
+      NearFieldEA_file << "\n";
+      for (iVertex = 0; iVertex < EquivArea_PhiAngle[0].size(); iVertex++) {
+
+        su2double XcoordRot = Xcoord_PhiAngle[0][iVertex]*cos(AoA) - Zcoord_PhiAngle[0][iVertex]*sin(AoA);
+        su2double XcoordRot_init = Xcoord_PhiAngle[0][0]*cos(AoA) - Zcoord_PhiAngle[0][0]*sin(AoA);
+
+        if (config->GetSystemMeasurements() == US)
+          NearFieldEA_file << scientific << (XcoordRot - XcoordRot_init) * 12.0;
+        else
+          NearFieldEA_file << scientific << (XcoordRot - XcoordRot_init);
+
+        for (iPhiAngle = 0; iPhiAngle < PhiAngleList.size(); iPhiAngle++) {
+          NearFieldEA_file << scientific << ", " << EquivArea_PhiAngle[iPhiAngle][iVertex];
+        }
+
+        NearFieldEA_file << "\n";
+
+      }
+      NearFieldEA_file.close();
+
+    }
+
+    /*--- Read target equivalent area from the configuration file,
+     this first implementation requires a complete table (same as the original
+     EA table). so... no interpolation. ---*/
+
+    vector<vector<su2double> > TargetArea_PhiAngle_Trans;
```

Comment:
su2activematrix for things that look like matrices

Reply:
`su2activematrix` doesn't appear to have a method equivalent to `push_back`. Do I have to resize the matrix and insert values one by one?

Reply:
Oh if it is using push_back you can leave it like it is.

Reply:
OK, I'll leave it as is then.